core/scripts/chaincli/handler/keeper_upkeep_history.go

```diff
@@ -0,0 +1,173 @@
+package handler
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"math/big"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/smartcontractkit/chainlink/core/services/keeper"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+)
+
+const (
+	defaultMaxBlocksRange = 1000
+)
+
+// UpkeepHistory prints the checkUpkeep status and keeper responsibility for a given upkeep in a set block range
+func (k *Keeper) UpkeepHistory(ctx context.Context, upkeepId int64, from, to uint64) {
```

Comment:
what happens if the upkeep is cancelled or not valid?

Reply:
```
 Block          checkUpkeep     Keeper Index    Keeper Address                                  Max LINK PaymentGas Limit       Adjusted Gas    LINK ETHPerform Data    Reason                  
 ----           ----            ----            ----                                            ----            ----            ----            ----    ----            ----                    
 7312030        false           6               0x09B2c50ddB8782EE07641D675Da8c9bc542b94df      <nil>           <nil>           <nil>           <nil>                   execution reverted      
 7312031        false           6               0x09B2c50ddB8782EE07641D675Da8c9bc542b94df      <nil>           <nil>           <nil>           <nil>                   execution reverted      
 7312032        false           6               0x09B2c50ddB8782EE07641D675Da8c9bc542b94df      <nil>           <nil>           <nil>           <nil>                   execution reverted      
 7312033        false           6               0x09B2c50ddB8782EE07641D675Da8c9bc542b94df      <nil>           <nil>           <nil>           <nil>                   execution reverted      
 7312034        false           6               0x09B2c50ddB8782EE07641D675Da8c9bc542b94df      <nil>           <nil>           <nil>           <nil>                   execution reverted      
 7312035        false           6               0x09B2c50ddB8782EE07641D675Da8c9bc542b94df      <nil>           <nil>           <nil>           <nil>                   execution reverted      
 ----           ----            ----            ----                                            ----            ----            ----            ----    ----            ----
```